weave/scorers/hallucination_scorer.py

```diff
@@ -157,3 +173,124 @@ async def score(self, output: str, context: str) -> HallucinationResponse:
             response.choices[0].message.content
         )
         return response.model_dump()
+
+
+HALLUCINATION_SCORER_THRESHOLD = 0.35
+
+
+class WeaveHallucinationScorerV1(HuggingFacePipelineScorer):
+    """
+    A scorer that detects hallucinations in the output, given an query and context. This scorer
+    uses the HHEM 2.1 model from Vectara, https://huggingface.co/vectara/hallucination_evaluation_model
+
+    This scorer uses a fine-tuned LLM to analyze whether model outputs contain information not supported
+    by the given context.
+
+    Args:
+        device: Device to run model on, defaults to "cuda"
+        model_name_or_path: Path or name of model weights to load
+
+    Note: This Scorer's `score` method will score the text passed to its `output` parameter for hallucinations
+    based on the query and context.
+
+    Returns:
+        dict: A dictionary containing:
+            - pass (bool): True if no hallucinations detected (score <= threshold)
+            - extras (dict): Contains:
+                - score (float): Hallucination score between 0 and 1, where lower is better
+                - error (str, optional): Error message if something went wrong
+
+    Example:
+        >>> scorer = HallucinationScorer()
+        >>> result = scorer.score(
+        ...     query="What is the capital of France?",
+        ...     context="Paris is the capital of France.",
+        ...     output="Paris is the capital of France."
+        ... )
+        >>> print(result)
+        {
+            'pass': True,
+            'extras': {
+                'score': 0.1
+            }
+        }
+    """
+
+    threshold: float = HALLUCINATION_SCORER_THRESHOLD
+    _model_max_length: int = PrivateAttr(default=8192)
+    task: str = "pair-classification"
```

Comment:
Move public attrs up, and add Field descriptions

Reply:
updated